main/modules/importer/scrivener/importer.js

```diff
@@ -628,27 +660,55 @@ function generateNotes(currentState, json, bookId, files, isNewFile) {
         content: fileContents.rtfContents,
       }
       createNewNote(currentState.notes, values)
+    } else if (children['BinderItem']) {
+      const binderItemsArr = Array.isArray(children['BinderItem'])
+        ? children['BinderItem']
+        : [children['BinderItem']]
+
+      generateNotes(currentState, binderItemsArr, bookId, files)
     }
   })
 }
 
 function generatePlaces(currentState, json, bookId, files, isNewFile) {
-  json.forEach((item) => {
+  json.forEach((item, key) => {
     const id = item['_attributes']['UUID'] || item['_attributes']['ID']
-    const name = item['Title']['_text']
+    const name =
+      item['Title'] && item['Title']['_text']
+        ? item['Title']['_text']
+        : t('Place {key}', { key: key + 1 })
+    const children = item['Children']
     const matchFiles = getMatchedRelevantFiles(files, id)
     if (matchFiles && matchFiles.length) {
       let fileContents = { txtContent: createSlateParagraph(''), rtfContents: [] }
-      const mappedFiles = mapMatchedFiles(currentState, matchFiles, fileContents, bookId)
+      const mappedFiles = mapMatchedFiles(currentState, matchFiles, fileContents, bookId, true)
       fileContents = {
         rtfContents: mappedFiles.rtfContents,
         txtContent: createSlateEditor(mappedFiles.txtContent),
       }
-      const content = generateCustomAttribute({ contents: fileContents.rtfContents }, name)
-      const customAttributes = content.reduce(
-        (obj, item) => Object.assign(obj, { [Object.keys(item)[0]]: Object.values(item)[0] }),
-        {}
+      const content = generateCustomAttribute(
+        { contents: fileContents.rtfContents },
+        currentState,
+        name
       )
+      const customAttributes = content
+        // remove three dots placeholder if key
+        .filter((item) => Object.keys(item)[0] != '...')
```

Comment:
Doing something like `Object.keys(x)[0]` is only valid if there's exactly one key on the object.
Otherwise, you're actually choosing a key potentially at random(!)
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#deleted_added_or_modified_properties

Reply:
Also see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values
(It says that `Object.values` -and it's the same for `Object.entries` and `Object.keys`- iterates over the object's properties in the same order as `for...in`.)

Reply:
Yes @Quiescent because it is an array of objects

Reply:
Then we might instead use `item[0]`, right?

Reply:
I get undefined 😅 inside the filter and map function when i use item[0]

Reply:
I mean the content its an array of objects, the item is an object 😅

Reply:
Right.  That's fine :+1: 
The problem here is the assumption that `Object.keys(item)[0]` will produce the correct key.
It might do that in your tests, but the standard doesn't guarantee this behaviour.
Does `item` have a known schema?

Reply:
@jeanawhou thanks, but we might be missing each other here.
I'm trying to resolve the problem caused by using `Object.keys(item)[0]` <--
Here, the intention seems to be to read the first key of an object.
What I'm saying is that `Object.keys(item)[0]` could give you *any* key(!)
So let's figure out what *`item`* looks like (i.e. what are its keys) so that we can avoid using `Object.keys` (because using `Object.keys` is probably a bug).

Reply:
This how an item looks (1st image)
<img width="1492" alt="image" src="https://user-images.githubusercontent.com/19387007/158351134-1ea01670-a44e-4ed3-a642-172d0727462b.png">

And it will depend on the custom attribute they wrote

on the scrivener
It will look llike this 
<img width="1493" alt="image" src="https://user-images.githubusercontent.com/19387007/158351657-72faa1da-4d0b-4cef-ad2b-fba2d187e9a0.png">

On the rtf it will look like this 
<img width="908" alt="image" src="https://user-images.githubusercontent.com/19387007/158351820-8b164c9d-26b3-4d59-86be-499f7511bcb2.png">

Reply:
Thanks for explaining :+1:  :slightly_smiling_face:
Ok.  So items are **by definition** single attribute objects(?)
I'm happy to keep it as is as long as we add a comment at the top of the function that makes the assumption explicit.
i.e. // ASSUMPTION: `item`s are single-key objects with a name that corresponds to an entity's property and a value which corresponds to that property's value.